### PR TITLE
Applying Cordova audio fixes.

### PR DESCRIPTION
**Pull request by yanshcherbakovatredspace**
This is to make Springroll Sound to be able to use Cordova Audio when required.

It will only be used when an environment specific objected is appended to the webview in order to override web audio and use native audio instead.

Such a use case is typically necessary when low latency audio is required and where web audio seems to be problematic. Typically on Android devices.

It is known that CordovaAudioPlugin is currently not working in iOS, therefore it should not be used with iOS devices.

CordovaAudioPlugin is used when the environment specific object `window.springroll.forceNativeAudio` is present. It is up to the Cordova application logic to append this object to the web view according to the supported device. In other words, if CordovaAudioPlugin should not be used on a certain device, the logic shall be determined in the Cordova application so that WebAudio is always the prevalent audio plugin.

### CODE REVIEW

File: src/media/Sound.js [hunk 1]
```diff
@@ -261,9 +265,20 @@
 
 			waitFunction = function()
 			{
-				window.top.plugins.NativeAudio.getCapabilities(function(result) {
-					waitResult = result;
-				});
+				//TODO: DRYify this.
+				if (window.plugins.NativeAudio) {
```
Comment (bigtimebuddy) [inline src/media/Sound.js]:
```js
var NativeAudio = window.plugins.NativeAudio || windows.top.plugins.NativeAudio || null;
if (NativeAudio) {
  NativeAudio.getCapabilities(function(result) {
    waitResult = result;
  });
}
```

Comment (yanshcherbakovatredspace) [inline src/media/Sound.js]: 💯

Comment (yanshcherbakovatredspace) [inline src/media/Sound.js]: thanks @bigtimebuddy

File: dist/modules/sound.js [hunk 2]
```diff
@@ -446,6 +446,7 @@
 		Debug,
 		SoundContext,
 		SoundInstance,
+		CordovaAudioPlugin = include('createjs.CordovaAudioPlugin'),
```
Comment (andrewstart) [inline dist/modules/sound.js]: The build integrations are failing on this - would that be solved by the corresponding SoundJS PR?

File: dist/modules/sound.js [hunk 3]
```diff
@@ -691,13 +698,21 @@
...
 				{
-					Application.instance.off("update", waitFunction);
-					_instance._initComplete(options.types, options.ready);
+					NativeAudio.getCapabilities(function(result)
+					{
+						waitResult = result;
+					});
 				}
+
+				Application.instance.off("update", waitFunction);
```
Comment (andrewstart) [inline dist/modules/sound.js]: Shouldn't this wait until `NativeAudio.getCapabilities` performs its callback?

File: src/media/Sound.js [hunk 4]
```diff
@@ -657,11 +672,6 @@
 				//fade the last played instance
 				inst = sound.playing[sound.playing.length - 1];
 			}
-			else if (s.loadState == LoadStates.loading)
```
Comment (andrewstart) [inline src/media/Sound.js]: Why was this removed?

File: dist/modules/sound.js (hunk 3 above)
Comment (yanshcherbakovatredspace) [inline dist/modules/sound.js]: Yes, you're right my bad. I think I intended that to be in call back block but moved it out. Unless I did that intentionally...

File: dist/modules/sound.js (hunk 2 above)
Comment (yanshcherbakovatredspace) [inline dist/modules/sound.js]: Yeah, I just had to change this line to `CordovaAudioPlugin = include('createjs.CordovaAudioPlugin', false),` and also bump the version in bower to take the latest SoundJS tag.

File: src/media/Sound.js (hunk 4 above)
Comment (yanshcherbakovatredspace) [inline src/media/Sound.js]: I believe this was removed in a previous merge.

Comment (yanshcherbakovatredspace) [inline src/media/Sound.js]: My bad... it was added in a previous commit but not merged into this branch.
